src/legacy/core_plugins/vis_type_timelion/common/lib/to_milliseconds.ts

```diff
@@ -33,26 +35,27 @@ const vals = _.transform(
     ['M', 'months', 'month'],
     ['quarter'],
     ['y', 'years', 'year'],
-  ],
-  function(vals, units) {
-    const normal = moment.normalizeUnits(units[0]);
+  ] as any,
+  // @ts-ignore
```

Comment:
I disabled TS here because `transform` accepts `accumulator` as an array, but in this case an object used. And to show the real types (`Values`) I had to disable TS.

![image](https://user-images.githubusercontent.com/11524637/72502521-9367a500-384a-11ea-97ae-e89f19aed595.png)

Reply:
Using transform is unnecessary here, this code is just iterating over the static input array and is putting stuff into an object. You can also write it like this:
```ts
const unitMappings = [
  ['ms', 'milliseconds', 'millisecond'],
  ['s', 'seconds', 'second', 'sec'],
  ['m', 'minutes', 'minute', 'min'],
  ['h', 'hours', 'hour'],
  ['d', 'days', 'day'],
  ['w', 'weeks', 'week'],
  ['M', 'months', 'month'],
  ['quarter'],
  ['y', 'years', 'year'],
] as Units[][];
const vals = {} as Values;
unitMappings.forEach(units => {
  const normal = moment.normalizeUnits(units[0]) as Units;
  const val = moment.duration(1, normal).asMilliseconds();
  ([] as Units[]).concat(normal, units).forEach((unit: Units) => {
    vals[unit] = val;
  });
});
```